frontend/src/views/instructor/routes/index.tsx

```diff
@@ -1,21 +1,46 @@
 import React from "react";
-import { Redirect, Route, Switch } from "react-router-dom";
+import { Redirect, Route, Switch, useParams } from "react-router-dom";
 import { InstructorAssignmentsView } from "../assignments";
 import { Landing as InstructorLanding } from "../landing";
+import { InstructorDdahsView } from "../ddahs";
+import { useThunkDispatch } from "../../../libs/thunk-dispatch";
+import { setActivePositionId } from "../store/actions";
+
+/**
+ * React component that will update the active position id in the
+ * Redux store whenever it is rendered. This component is meant to be the first
+ * component in a route that depends on an active position.
+ */
+function UpdateActivePosition() {
+    const params = useParams<{ position_id?: string }>();
+    const dispatch = useThunkDispatch();
+    const rawPositionId = +(params.position_id || NaN);
+    const position_id = Number.isFinite(rawPositionId) ? rawPositionId : null;
```

Comment:
Can we change it to be something more readable? For example,

```
const rawPositionId = params.position_id;
const position_id = typeof rawPositionId === 'number' ? rawPositionId : null;
```

or 

```
const rawPositionId = params.position_id;
const position_id = isNan(rawPositionId) ? null : rawPositionId;
```

Reply:
Well, it starts out as a string. `isNan` could be used, but that lets `Infinity` through...